Mathlib/Algebra/MvPolynomial/Basic.lean

```diff
@@ -645,6 +645,20 @@ def coeffAddMonoidHom (m : σ →₀ ℕ) : MvPolynomial σ R →+ R
   map_add' := coeff_add m
 #align mv_polynomial.coeff_add_monoid_hom MvPolynomial.coeffAddMonoidHom
 
+variable (R)
+/-- `MvPolynomial.coeff m` but promoted to a `LinearMap`. -/
+def lcoeff (m : σ →₀ ℕ) : MvPolynomial σ R →ₗ[R] R where
+  toFun := coeff m
+  map_add' := coeff_add m
+  map_smul' := coeff_smul m
+
+variable {R}
+
+@[simp]
+theorem lcoeff_apply (m : σ →₀ ℕ) (f : MvPolynomial σ R) :
+    lcoeff R m f = coeff m f :=
+  rfl
```

Comment:
Untested:
```suggestion
variable (R) in
/-- `MvPolynomial.coeff m` but promoted to a `LinearMap`. -/
@[simps apply]
def lcoeff (m : σ →₀ ℕ) : MvPolynomial σ R →ₗ[R] R where
  toFun := coeff m
  map_add' := coeff_add m
  map_smul' := coeff_smul m
```

Reply:
Yes, that works, as pointed out by @erdOne in the comments to #12293.  
This file is common to the 3 PR, I did not take the time to sort out which of these modifications were necessary. I'm sorry for the useless work.

Reply:
If these PRs share commen code and you don't want to spend time to extract the dependency, can you mark them as dependencies in some arbitrary order to avoid further duplicate work?

Reply:
I'll do that, sorry. 
Let's make the dependency in the order they are.

Reply:
So Lean creates the _apply lemma, but it has R as an explicit argument. 
This is not such a nuisance, I believe. Maybe we should be satisfied with this. 
(Maybe `simps` has to be refined so as to put implicit arguments when desired.)